src/run-tests-in-page.js

```diff
@@ -0,0 +1,28 @@
+const puppeteer = require("puppeteer");
+const runTestcases = require("./run-testcases");
+const { concatReport } = require("./axe-reporter-earl");
+
+async function runTestsInPage(program, toolRunner) {
+  const { page, browser } = await puppeteerSetup();
+  const testResults = await runTestcases(program, testcase => {
+    return toolRunner(page, testcase);
+  });
+
+  await stopPuppeteer({ page, browser });
+  return concatReport(testResults);
+}
+
+async function puppeteerSetup() {
```

Comment:
given the other function is called `stopPuppeteer`, perhaps this should be named `setupPuppeteer`.

On second thought, both of these functions can be inline within `runTestsInPage`. Having these a separate fn's makes it hard to read.